test/unittests/execute_floating_point_test.cpp

```diff
@@ -66,4 +66,129 @@ TEST(execute_floating_point, f64_add)
+    constexpr auto lt = func_offset + 2;
+    constexpr auto gt = func_offset + 3;
+    constexpr auto le = func_offset + 4;
+    constexpr auto ge = func_offset + 5;
+
+    const TypeParam ordered_values[] = {
+        -Limits::infinity(),
+        -Limits::max(),
+        std::nextafter(-Limits::max(), TypeParam{0}),
+        std::nextafter(-TypeParam{1.0}, -Limits::infinity()),
+        -TypeParam{1.0},
+        std::nextafter(-TypeParam{1.0}, TypeParam{0}),
+        std::nextafter(-Limits::min(), -Limits::infinity()),
+        -Limits::min(),
+        std::nextafter(-Limits::min(), TypeParam{0}),
+        std::nextafter(-Limits::denorm_min(), -Limits::infinity()),
+        -Limits::denorm_min(),
+        TypeParam{0},  // FIXME.
+        Limits::denorm_min(),
+        std::nextafter(Limits::denorm_min(), Limits::infinity()),
+        std::nextafter(Limits::min(), TypeParam{0}),
+        Limits::min(),
+        std::nextafter(Limits::min(), Limits::infinity()),
+        std::nextafter(TypeParam{1.0}, TypeParam{0}),
+        TypeParam{1.0},
+        std::nextafter(TypeParam{1.0}, Limits::infinity()),
+        std::nextafter(Limits::max(), TypeParam{0}),
+        Limits::max(),
+        Limits::infinity(),
+
+        // NaNs.
+        FP<TypeParam>::nan(FP<TypeParam>::canon),
+        FP<TypeParam>::nan(FP<TypeParam>::canon + 1),
+        FP<TypeParam>::nan(1),
+    };
+
+    // Check every pair from cartesian product of ordered_values.
+    for (size_t i = 0; i < std::size(ordered_values); ++i)
+    {
+        for (size_t j = 0; j < std::size(ordered_values); ++j)
+        {
+            const auto a = ordered_values[i];
+            const auto b = ordered_values[j];
+            if (std::isnan(a) || std::isnan(b))
+            {
+                EXPECT_THAT(execute(*inst, eq, {a, b}), Result(0)) << a << "==" << b;
+                EXPECT_THAT(execute(*inst, ne, {a, b}), Result(1)) << a << "!=" << b;
+                EXPECT_THAT(execute(*inst, lt, {a, b}), Result(0)) << a << "<" << b;
+                EXPECT_THAT(execute(*inst, gt, {a, b}), Result(0)) << a << ">" << b;
+                EXPECT_THAT(execute(*inst, le, {a, b}), Result(0)) << a << "<=" << b;
+                EXPECT_THAT(execute(*inst, ge, {a, b}), Result(0)) << a << ">=" << b;
+            }
+            else
+            {
+                EXPECT_THAT(execute(*inst, eq, {a, b}), Result(uint32_t{i == j})) << a << "==" << b;
+                EXPECT_THAT(execute(*inst, ne, {a, b}), Result(uint32_t{i != j})) << a << "!=" << b;
+                EXPECT_THAT(execute(*inst, lt, {a, b}), Result(uint32_t{i < j})) << a << "<" << b;
+                EXPECT_THAT(execute(*inst, gt, {a, b}), Result(uint32_t{i > j})) << a << ">" << b;
+                EXPECT_THAT(execute(*inst, le, {a, b}), Result(uint32_t{i <= j})) << a << "<=" << b;
+                EXPECT_THAT(execute(*inst, ge, {a, b}), Result(uint32_t{i >= j})) << a << ">=" << b;
+            }
+        }
+    }
+
+    // Negative zero.
```

Comment:
Why are these needed if we have `-0` and `0` in `ordered_values`?

Reply:
We don't.

Reply:
Ah, its -1 and 0. Can we just have -0 there too?

Reply:
No.

Reply:
Why?

Reply:
Why?

Reply:
The values in the array must be strictly ordered, otherwise we don't know what result to expect. But `-0 == 0`.

Reply:
I don't see why they need to be strictly ordered, to get the expected result you always calculate it as `i op j` anyway. Also as at some iterations `i < j`, at others `j < i` (and at some `i == j`)

Reply:
Please commit a change doing that because I don't see how this suppose to work.

Reply:
Oh I see, I misunderstood how it works first. But I pushed the way I thought it works first, it doesn't rely on strict ordering, what do you think?